drivers/adc/ad7616/ad7616.c

```diff
@@ -909,10 +911,12 @@ int32_t ad7616_self_test(struct ad7616_dev *dev)
 	if (ret != 0)
 		return ret;
 
-	// First toggle to remove pending garbage
-	ret = ad7616_toggle_conv(dev);
-	if (ret != 0)
-		return ret;
+	// First couple of toggles return garbage values, discard them
```

Comment:
any idea why this is happening? there should be an explanation from the data-sheet here

Reply:
The first one is expected when reading the datasheet. But when I compile the project in release mode (-O2) the second one is also not correct.
On a side note a similar issue exist on the `ad7616_sdz` where the first 4 values are not correct (either 0 or value that does not correspond to the signal being sampled)

Reply:
it sounds like a timing issue, I think it should be investigated what is the root-cause of this... does adding a delay after a single ad7616_toggle_conv call produce similar results?

Reply:
> it sounds like a timing issue

I agree.

> does adding a delay after a single ad7616_toggle_conv call produce similar results?

I tested that without success at some point. The configuration is a bit different now so I will test again

Reply:
I'm not able to reproduce the issue at the moment. I definitely had the issue in previous versions where without this fix it was working 3 out of 4 times.

As I can't reproduce the bug at the moment I think I can remove the fix for now but I would not be surprised if we encounter this issue again

Reply:
I would say to remove the fix. As long as we do not know the root-cause (and cannot reproduce the issue currently) we can not say for sure that this is the fix, it might do the trick for your setup, but for another setup it might not.